c10/core/SymIntArrayRef.cpp

```diff
@@ -28,7 +28,11 @@ at::IntArrayRef asIntArrayRefUnchecked(c10::SymIntArrayRef ar) {
 
 // TODO: this print is bad
 std::ostream& operator<<(std::ostream& os, SymInt s) {
-  os << "SymInt(" << s.as_int_unchecked() << ")";
+  if (s.is_symbolic()) {
+    os << "SymInt(" << s.toSymIntNodeImpl()->str() << ")";
```

Comment:
is it safe enough to assume the node is non-null?

Reply:
yeah, that should be an invariant

Reply:
SymInt can default initialize to non-symbolic zero